src/SwaggerProvider.DesignTime/OperationCompiler.fs

```diff
@@ -99,36 +108,52 @@ type OperationCompiler (schema:SwaggerObject, defCompiler:DefinitionCompiler) =
                 let obj = Expr.Coerce(exp, typeof<obj>)
                 <@ (%%obj : obj).ToString() @>
 
-            let rec corceQueryString name expr =
+            let rec coerceQueryString name expr =
                 let obj = Expr.Coerce(expr, typeof<obj>)
                 <@ let o = (%%obj : obj)
                    SwaggerProvider.Internal.RuntimeHelpers.toQueryParams name o @>
 
-            let replacePathTemplate path name (exp : Expr) =
-                let template = "{" + name + "}"
-                <@@ Regex.Replace(%%path, template, string (%%exp : string)) @@>
+            let replacePathTemplate path name (exp: Expr<string>) =
+                let template = sprintf "{%s}" name
+                <@ Regex.Replace(%path, template, %exp) @>
 
+            let createStream (stringData: string) = 
+                stringData |> Text.Encoding.UTF8.GetBytes |> MemoryStream :> Stream
+                
             let addPayload load (param : ParameterObject) (exp : Expr) =
                 let name = param.Name
-                let var = coerceString param.Type param.CollectionFormat exp
+                // delay the string coercion so that if it's a stream we don't tostring it.
+                let var () = coerceString param.Type param.CollectionFormat exp
                 match load with
-                | Some (FormData, b) -> Some (FormData, <@@ Seq.append %%b [name, (%var : string)] @@>)
-                | None               -> match param.In with
-                                        | Body -> Some (Body, Expr.Coerce (exp, typeof<obj>))
-                                        | _    -> Some (FormData, <@@ (seq [name, (%var : string)]) @@>)
+                | Some (BodyForm f) -> 
+                    Some (BodyForm <@ Seq.append %f [name, %var()] @>)
+                | Some (BodyForm f) when param.Type = SchemaObject.File -> 
```

Comment:
if we get a file param, now we have to switch to multipart.  for this we have to convert the previous entries into the contentType*fileName*data structure that files have